### ISSUE TITLE
[UX/ScriptNode] Duplicating/copypasting nodes severs the cable connection to parent/master container parameters

### ISSUE THREAD

**Issue by aaronventure** [christophhart/HISE#608]
Understandable for container parameter cables being severed when copypasting if the severing happens on copy, as you don't know whether the nodes are going to get pasted within the same container. Although that could be decided on paste. 

For duplication it should not sever any parameter cables, as these can only be from parent containers (duplicated node always ends up in the same container).

Master container parameter connections should never be severed.

**Comment by christoph-hart** [comment]
Yeah, the problem is that I'm using the same logic for duplicating nodes within a container (where it might be possible to keep parent parameter connections) and copy / pasting between networks (where it isn't possible) and I'd love to avoid branching that because it will definitely break on the next round of refactoring.

**Comment by aaronventure** [comment]
Is it a function? Won't adding an argument isDuplication and setting this to true for the Ctrl+D call let you easily and cleanly branch this? 

I understand that copy and paste might not get this as this would include adding logic on paste (if it doesn't exist). In that case it would probably be
* check if parent with the same id exists
* check if parent has the same parameter ids

This is enough safeguarding that it would have to be intentional to happen if copying to another network.
